flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/BaseDeltaTaskWriter.java

```diff
@@ -74,11 +76,18 @@ public void write(RowData row) throws IOException {
       case INSERT:
       case UPDATE_AFTER:
         if (upsert) {
-          writer.delete(row);
+          writer.deleteKey(row);
```

Comment:
`RowDataProjection.create(schema, deleteSchema).wrap(row)`

Reply:
Hmmm that didn't work as currently used. But changing the abstract method higher up did.